src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1541,6 +1588,66 @@ Node* VectorLongToMaskNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   return NULL;
 }
 
+// Generate other vector nodes to implement the masked/non-masked vector negation.
+Node* VectorNode::degenerate_vector_integral_negate(Node* n, int vlen, BasicType bt, PhaseGVN* phase, bool is_predicated) {
```

Comment:
Shall we move this declaration in `class  NegVNode` since it is only used by  NegVNode::Ideal ?

Reply:
I think it can be. Thanks for pointing out this. I will change this later.